pkg/apis/build/v1alpha1/build_validation_test.go

```diff
@@ -38,6 +38,369 @@ func TestValidateBuild(t *testing.T) {
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+		reason: "sources with prefixed targetPath",
+	}, {
+		desc: "sources with overlap of targetPath",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name:       "testab",
+					TargetPath: "/path/a/b",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}, {
+					Name:       "testa",
+					TargetPath: "/path/a",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+		reason: "sources with similar subpath for targetPath",
+	}, {
+		desc: "sources with overlap of targetPath in different order",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name:       "testab",
+					TargetPath: "/path/a",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}, {
+					Name:       "testa",
+					TargetPath: "/path/a/b",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+		reason: "sources with nested paths for targetPath",
+	}, {
+		desc: "sources with combination of empty and non empty targetPath",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name:       "gitpathab",
+					TargetPath: "/path/a/b",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}, {
+					Name: "gitpathad",
+					GCS: &GCSSourceSpec{
+						Type:     GCSArchive,
+						Location: "blah",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+	}, {
+		desc: "sources with empty targetPaths",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name:       "gitpathab",
+					TargetPath: "/path/a/b",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}, {
+					Name: "gcsnopath1",
+					GCS: &GCSSourceSpec{
+						Type:     GCSArchive,
+						Location: "blah",
+					},
+				}, {
+					Name: "gcsnopath", // 2 sources with empty target path
+					GCS: &GCSSourceSpec{
+						Type:     GCSArchive,
+						Location: "blah",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+		reason: "multiple sources with empty target paths",
+	}, {
+		desc: "sources with combination of different targetPath with common parent dir",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name:       "gitpathab",
+					TargetPath: "/a/b/d",
```

Comment:
Should this error? What's the expected result of this?

Reply:
I moved all path related test cases into target_path_test.go. Let me know if I covered this case